webrtc.html

```diff
@@ -4074,90 +4194,123 @@ <h3 id="sec.identity-verify-assertion">Verifying Identity Assertions</h3>
+        </li>
+        <li>
+          <p>The <code>RTCPeerConnection</code> instantiates the identified IdP
+          as described in <a href="#sec.identity-proxy-communications"></a> and
+          <a href="#sec.register-idp"></a>.  If the IdP cannot be loaded,
+          instantiated or the IdP proxy is not registered, this process
+          fails.</p>
         </li>
 
-        <li>The <code>RTCPeerConnection</code> sends a "VERIFY" message to the
-        IdP proxy. This message includes the assertion from the offer/answer
-        which is to be verified.</li>
-
-        <li>The IdP proxy verifies the identity assertion (depending on the
-        authentication protocol this could involve interacting with the IDP
-        server).</li>
+        <li>
+          <p>The <code>RTCPeerConnection</code> invokes the <code><a
+          href="#widl-RTCIdentityProvider-validateAssertion-Promise-RTCIdentityValidationResult--DOMString-assertion-DOMString-origin">validateAssertion</a></code>
+          method on the <code><a>RTCIdentityProvider</a></code> instance
+          registered by the IdP.</p>
+
+          <p>The <var>assertion</var> parameter is taken from the decoded
+          identity assertion.  The <var>origin</var> parameter contains the
+          origin of the script that calls the <code>RTCPeerConnection</code>
+          method that triggers this behavior.
+        </li>
 
-        <li>Once the assertion is verified, the IdP proxy sends a response
-        containing the verified assertion results to the
-        <code>RTCPeerConnection</code> over the message channel.</li>
+        <li>
+          <p>The IdP proxy returns a promise and performs the validation process
+          asynchronously.</p>
+          <p>The IdP proxy verifies the identity assertion using whatever means
+          necessary.  Depending on the authentication protocol this could
+          involve interacting with the IDP server.</p>
+        </li>
 
-        <li>The <code>RTCPeerConnection</code> validates that the fingerprint
-        provided by the IdP in the validation response matches the certificate
-        fingerprint that is, or will be, used for communications. This is
-        either by:
+        <li>
+          <p>Once the assertion is successfully verified, the IdP proxy resolves
+          the promise with an <code><a>RTCIdentityValidationResult</a></code>
+          containing the validated identity and the original contents that are
+          the payload of the assertion.</p>
+        </li>
 
-          <ul>
-            <li>Ensuring that there is only a single a=fingerprint value across
-            all media sections in the SDP that matches the fingerprint provided
-            by the IdP.</li>
+        <li>
+          <p>The <code>RTCPeerConnection</code> decodes the contents and
+          validates that it contains at least one fingerprint value for each
+          <code>a=fingerprint</code> attribute in the session description.  This
```

Comment:
Yeah, that's overly precise.  Our code permits multiples (and ignores them) by virtue of doing a lookup.